### PR TITLE
Add basic tests to other benchmarks

### PR DESCRIPTION
**Pull request by peterrum**
There are two problematic benchmarks and this PR depends on #11427. Nevertheless, I think we can merge.

### CODE REVIEW

File: benchmark_precond_merged/test.cc [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+
+
+int
+main(int argc, char **argv)
+{
+  Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
+
+  // run();
```
Comment (peterrum) [inline benchmark_precond_merged/test.cc:35]: For some reason this version crashes with NaN.

File: benchmark_s_step/test.cc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+
+
+int
+main(int argc, char **argv)
+{
+  Utilities::MPI::MPI_InitFinalize mpi(argc, argv, 1);
+
+  return 0;
```
Comment (peterrum) [inline benchmark_s_step/test.cc]: And this one too (I have not figured out yet why).

File: common_code/benchmark_test.h [hunk 3]
```diff
@@ -249,7 +249,8 @@ do_test(ConvergenceTable &table)
                       &comm_shmem);
 #endif
 
-  unsigned int s = 1 + std::log2(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD));
+  unsigned int s =
+    std::max<unsigned int>(min_s, 1 + std::log2(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD)));
```
Comment (peterrum) [inline common_code/benchmark_test.h:253]: I had to limit the minimum size for s-step CG...

File: benchmark_s_step/test.mpirun=1.output [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  3      2  6      64   2187   17 1.1560e-01 
+  3      2  7     128   4131   19 1.1554e-01 
+  3      2  8     256   7803   24 1.1506e-01 
+  3      2  9     512  14739   34 2.1551e-02 
+  3      3  5      32   3549   22 2.6666e-01 
+  3      3  6      64   6591   29 3.9731e-02 
+  3      3  7     128  12675   32 3.9734e-02 
+  3      3  8     256  24375   41 3.9586e-02 
+dim degree s  n_cells n_dofs n_it   error    
+  3      1  5      32    225    2 5.3575e-01
```
Comment (peterrum) [inline benchmark_s_step/test.mpirun=1.output:36]: The tables look good :)

**Review by kronbichler — APPROVED** [review]

**Pull request merged**